test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -263,24 +259,52 @@ pollWithRetry
     :: PactTestApiCmds
     -> ClientEnv
     -> RequestKeys
-    -> IO (Either ClientError PollResponses)
+    -> IO PollResponses
 pollWithRetry cmds env rks = do
   sleep 3
   go maxPollRetries
     where
-      go retries = do
-          result <- runClientM (pollApiCmd cmds (Poll (_rkRequestKeys rks))) env
+      go !retries = do
+          let retry failure = do
+                if retries == 0
+                  then do
+                    putStrLn $ "poll failing after " ++ show maxSendRetries
+                               ++ " retries"
+                    fail failure
+                  else sleep 1 >> go (retries - 1)
+          let rkks = _rkRequestKeys rks
+          result <- runClientM (pollApiCmd cmds (Poll rkks)) env
           case result of
-              Left _ ->
-                  if retries == 0 then do
-                      putStrLn $ "poll failing after " ++ show maxSendRetries ++ " retries"
-                      return result
-                  else do
-                      sleep 1
-                      go (retries - 1)
-              Right _ -> do
-                  putStrLn $ "poll succeeded after " ++ show (maxSendRetries - retries) ++ " retries"
-                  return result
+              Left e -> retry (show e)
+              Right (PollResponses mp) -> do
+                  ok <- checkResults (toList rkks) mp
+                  if ok
+                    then do
+                        putStrLn $ concat
+                            [ "poll succeeded after "
+                            , show (maxSendRetries - retries)
+                            , " retries" ]
+                        return $ PollResponses mp
+                    else do
+                        putStrLn "poll check failed, continuing"
```

Comment:
If these messages are needed for debugging, maybe put them behind some like:

```
#ifdef DEBUG_TESTS
debug = putStrLn
#else
debug = return ()
#endif
```

`DEBUG_TEST` could be define via the already existing [`debug` cabal flag](https://github.com/kadena-io/chainweb-node/blob/95ac85e95c5a8bd8ce17876bdc0ff44b7c562184/chainweb.cabal#L66).